src/librustdoc/html/render/mod.rs

```diff
@@ -2532,6 +2534,16 @@ fn compare_impl<'a, 'b>(lhs: &'a &&Impl, rhs: &'b &&Impl) -> Ordering {
     compare_names(&lhs, &rhs)
 }
 
+fn compare_impl_polarity(a: &Impl, b: &Impl) -> Ordering {
```

Comment:
You only take into account the polarity and not the name for the sorting, which isn't great... In case both impls have the same polarity, please return the comparison of the traits' name.

Reply:
Note that stable sorts are composable. If you first sort by property A  and then by property B then the sorting for A will serve as tie-breakers where B-comparisons return Ordering::Equal.